timeseries/src/autogluon/timeseries/utils/features.py

```diff
@@ -284,3 +296,63 @@ def _check_required_columns_are_present(
             raise ValueError(
                 f"{len(missing_columns)} columns are missing from {data_frame_name}: {reprlib.repr(missing_columns.to_list())}"
             )
+
+
+class AbstractFeatureImportanceTransform:
+    """Abstract class for transforms that replace a given feature with dummy or shuffled values,
+    for use in feature importance operations.
+    """
+
+    def __init__(
+        self,
+        forecast_target: str,
+        covariate_metadata: CovariateMetadata,
+    ):
+        self.forecast_target = forecast_target
+        self.covariate_metadata: CovariateMetadata = covariate_metadata
+
+    @classmethod
+    def from_feature_generator(
+        cls, feature_generator: TimeSeriesFeatureGenerator
+    ) -> "AbstractFeatureImportanceTransform":
+        return cls(
+            forecast_target=feature_generator.target,
+            covariate_metadata=feature_generator.covariate_metadata,
+        )
+
+    def _apply_transform(self, data: TimeSeriesDataFrame, target_feature: str) -> TimeSeriesDataFrame:
+        raise NotImplementedError
+
+    def transform(self, data: TimeSeriesDataFrame, target_feature: str, **kwargs) -> TimeSeriesDataFrame:
```

Comment:
Maybe rename `target_feature` -> `feature_name` to avoid confusion with the target time series?